### ISSUE TITLE
Error with flow type and argument desctructuring

### ISSUE THREAD

**Issue by danez** [babel/babel-eslint#239]
I reported this to eslint first in eslint/eslint#5021 but they told me to report it here.

Assuming the following code snippet:
```js
type Props = { baseUrl: string };

export default ({ baseUrl, anchor, inspectUtilities }: Props) => {
  return null;
};
```

eslint reports 1 error:

```
  3:56  error  "Props" is already declared in the upper scope  no-shadow
```

There should be no warnings, as both babel and flow correctly work with this typeannotation.

---

eslint version: v1.10.3
plugins: react, flow-vars
os: mac os x 10.11

---

.eslintrc:
```json
{
  "extends": "airbnb",
  "parser": "babel-eslint",
  "plugins": [
    "react",
    "flow-vars"
  ]
}
```

**Comment by danez** [comment]
Fixed by #246

**Comment by hzoo** [comment]
Thanks!
